### PR TITLE
#167727753 Pre-screened accommodation supplier can add accommodation

### PR DESCRIPTION
**Pull request by tobslob**
## What does this PR do?

PR that enable only pre-screened accommodation supplier to create accommodation

## Description of Task to be completed?

- create accommodation-supplier role
- update create accommodation controller
- seed accommodation supplier user
- test create accommodation endpoint
- update accommodation route

## How should this be manually tested?
- CLONE this repo
- open the cloned directory
- run ``` npm install ```
- run ``` npm run start:dev ```
- access the docs using {base-url}/docs
- POST accommodation using {base-url}/accommodation

## What are the relevant pivotal tracker stories?

[Finishes #167727753](https://www.pivotaltracker.com/story/show/167727753)

### CODE REVIEW

File: src/controllers/accommodationController.js [hunk 1]
```diff
@@ -17,8 +18,16 @@ const { serverError } = messages;
  * @returns {Object} - custom response
 */
 const createAccommodation = async (req, res) => {
+  const { TRAVEL_ADMIN, SUPER_ADMIN, ACCOMMODATION_SUPPLIER } = roles;
+  const checkRole = TRAVEL_ADMIN || SUPER_ADMIN || ACCOMMODATION_SUPPLIER;
```
Comment (tejiri4) [inline src/controllers/accommodationController.js]: LET TRY USING AUTHORIZE MIDDLEWARE ONLY AT FIRST AND SEE IF IT STILL WORKS

Comment (tobslob) [reply src/controllers/accommodationController.js]: It works

File: src/models/accommodation.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ module.exports = (sequelize, DataTypes) => {
       allowNull: false,
     },
     accommodationType: {
-      type: DataTypes.ARRAY(DataTypes.STRING),
+      type: DataTypes.STRING,
```
Comment (tejiri4) [inline src/models/accommodation.js:50]: Why did you changed  the data type?

**Comment by tejiri4** [comment]
Do look into the comment I dropped and Also rebase

File: src/models/accommodation.js (hunk 2 above)
Comment (tobslob) [reply src/models/accommodation.js:50]: I realized it should be a single type... depending on the type the vendor wants to provide

**Comment by tejiri4** [comment]
Try your PR on heroku using the review app and add a comment if your endpoint is working on heroku

**Review by tejiri4 — APPROVED** [review]
LGTM

**Pull request merged**